webapp/app/routes/app/l2-cancellations/new/_route.tsx

```diff
@@ -0,0 +1,215 @@
+  }),
+  refundRecipient: addressSchema,
+  txMintValue: z.coerce.number({ message: "Transaction mint value must be a number" }),
+});
+type Schema = z.infer<typeof schema>;
+
+const defaultValues = {};
+
+export default function NewL2GovernorVeto() {
+  const { activeL2Proposals } = useLoaderData<typeof loader>();
+  const form = useForm<Schema>({
+    resolver: zodResolver(schema),
+    defaultValues,
+    mode: "onTouched",
+  });
+  const navigation = useNavigation();
+
+  return (
+    <div>
+      <Form {...form}>
+        <form method="POST" className="space-y-4">
+          <Card className="pb-10">
+            <CardHeader>
+              <CardTitle>1. Select an active proposal</CardTitle>
+            </CardHeader>
+            <CardContent>
+              {activeL2Proposals.length > 0 && (
+                <Table>
+                  <TableHeader>
+                    <TableRow>
+                      <TableHead>ID</TableHead>
+                      <TableHead>Type</TableHead>
+                      <TableHead>Description</TableHead>
+                      <TableHead />
+                    </TableRow>
+                  </TableHeader>
+                  <TableBody>
+                    {activeL2Proposals.map((row) => (
+                      <TableRow key={row.proposalId}>
+                        <TableCell>{row.proposalId}</TableCell>
+                        <TableCell>{row.type}</TableCell>
+                        <TableCell>{row.description}</TableCell>
+                        <TableCell>
+                          <FormField
+                            control={form.control}
+                            name="proposalId"
+                            render={({ field }) => (
+                              <FormItem>
+                                <FormControl>
+                                  <Input type={"radio"} {...field} value={row.proposalId} />
+                                </FormControl>
+                              </FormItem>
+                            )}
+                          />
+                        </TableCell>
+                      </TableRow>
+                    ))}
+                  </TableBody>
+                </Table>
+              )}
+              {activeL2Proposals.length === 0 && (
+                <div className="text-center text-gray-500">No active proposals found.</div>
+              )}
+            </CardContent>
+          </Card>
+
+          <Card>
+            <CardHeader>
+              <CardTitle>2. Fill in the details for the veto proposal</CardTitle>
+            </CardHeader>
+            <CardContent className="space-y-4">
+              <FormField
+                control={form.control}
+                name="l2GasLimit"
+                render={({ field }) => (
+                  <FormItem>
+                    <FormLabel>L2 Gas Limit</FormLabel>
+                    <FormControl>
+                      <Input type="number" min={1} {...field} />
```

Comment:
Why is the minimum 1? Is it something enforced by the contracts?

Reply:
It's not, but I figured it couldn't be 0 or negative. It also can't be a float.